Foundation/Calendar.swift

```diff
@@ -862,8 +862,9 @@ public struct Calendar : Hashable, Equatable, ReferenceConvertible, _MutableBoxi
     public func hash(into hasher: inout Hasher) {
         // We implement hash ourselves, because we need to make sure autoupdating calendars have the same hash
         if _autoupdating {
-            hasher.combine(1 as Int8)
+            hasher.combine(false)
         } else {
+            hasher.combine(true)
```

Comment:
The original implementation here was ever so slightly wrong -- it had a variable-length hash encoding without discriminators to mark its boundaries, which could (theoretically) lead to collisions when multiple `Calendar` values are fed into a hasher.

Granted, I doubt this would be a problem in practice; however, this change makes `Calendar`'s hash encoding match that of the Foundation overlay, which is nice. I made similar changes for `TimeZone` and `Locale`.